androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/common/PrimaryTextField.kt

```diff
@@ -0,0 +1,86 @@
+package co.nimblehq.avishek.phong.kmmic.android.ui.common
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.*
+import androidx.compose.material.MaterialTheme.shapes
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.input.*
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+private val TextFieldHeight = 56.dp
+private const val PlaceholderColorAlpha = 0.3f
+
+@Composable
+fun PrimaryTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    placeholder: String,
+    modifier: Modifier = Modifier,
+    visualTransformation: VisualTransformation = VisualTransformation.None,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    imeAction: ImeAction = ImeAction.Next,
+    contentDescription: String = "",
+    singleLine: Boolean = true,
+) {
+    val focusManager = LocalFocusManager.current
+    TextField(
+        value = value,
+        onValueChange = onValueChange,
+        placeholder = {
+            Text(
+                text = placeholder,
+                style = MaterialTheme.typography.body1
+            )
+        },
+        singleLine = singleLine,
+        colors = appTextFieldColors(),
+        visualTransformation = visualTransformation,
+        keyboardOptions = KeyboardOptions(
+            keyboardType = keyboardType,
+            imeAction = imeAction
+        ),
+        keyboardActions = KeyboardActions(onDone = { focusManager.clearFocus() }),
+        modifier = modifier
+            .fillMaxWidth()
+            .heightIn(min = TextFieldHeight)
+            .clip(shapes.large)
+            .semantics { this.contentDescription = contentDescription }
+    )
+}
+
+@Composable
+internal fun appTextFieldColors(
```

Comment:
We can just use `private` here can't we?
```suggestion
private fun appTextFieldColors(
```

Reply:
@kaungkhantsoe the idea is to make it reusable. Suppose we need to update only the text color on some screen, we can call this method with only the updated text color.

Reply:
Please remind me how can we reuse it if it is `internal` function 🙏 . Also currently `PrimaryTextField` doesn't have options to change `colors` since it doesn't have a parameter for it.

Reply:
@kaungkhantsoe `internal` was used to make it visible only in the `:androidApp` module.
We don't have any other text fields at the moment, but if we do in some places, where one or some of the colors might be different from the rest of the app, it'll come in handy. Here's an example:
![image](https://user-images.githubusercontent.com/8093908/235621166-71ab9d5b-de21-45d7-a132-1d67aef53c7b.png)
Note: in real-life, the new text field will be in a separate composable.